lib/compress/zstd_compress.c

```diff
@@ -1250,20 +1250,30 @@ void ZSTD_seqToCodes(const seqStore_t* seqStorePtr)
         mlCodeTable[seqStorePtr->longLengthPos] = MaxML;
 }
 
-MEM_STATIC symbolEncodingType_e ZSTD_selectEncodingType(FSE_repeat* repeatMode,
-        size_t const mostFrequent, size_t nbSeq, U32 defaultNormLog)
+typedef enum {
+    ZSTD_defaultDisallowed = 0,
+    ZSTD_defaultAllowed = 1
+} ZSTD_defaultPolicy_e;
+
+MEM_STATIC symbolEncodingType_e ZSTD_selectEncodingType(
+        FSE_repeat* repeatMode, size_t const mostFrequent, size_t nbSeq,
+        U32 defaultNormLog, ZSTD_defaultPolicy_e const defaultAllowed)
 {
 #define MIN_SEQ_FOR_DYNAMIC_FSE   64
 #define MAX_SEQ_FOR_STATIC_FSE  1000
 
-    if ((mostFrequent == nbSeq) && (nbSeq > 2)) {
+    if ((mostFrequent == nbSeq) && (!defaultAllowed || nbSeq > 2)) {
```

Comment:
This test result is tied to `ZSTD_defaultAllowed==1`. 
Is it worthwhile to make it a static assert ?

Reply:
I'm not sure I understand what you mean. `defaultAllowed` is dynamic for offsets, since we check if `max <= DefaultMaxOff` value.

Reply:
I meant `ZSTD_defaultAllowed` as an `enum` value,
not `defaultAllowed` as a variable name.
Both are indeed spelled pretty close.

You are testing `!defaultAllowed`, which implies that `ZSTD_defaultAllowed==1`.
Since implicit assumption could be broken in the future, or just not clearly understood by a future reviewer, you could go being explicit with `defaultAllowed==ZSTD_defaultAllowed` instead.
(btw, I think I would prefer something like `isDefaultAllowed` as the variable name.)

I understand it's technically the same.
It's just about being more readable and reducing side effects in case of future changes.